docs/use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -302,98 +313,237 @@ $ script -fec 'sudo -E USE_DOCKER=true ./image_builder.sh ${ROOTFS_DIR}'
+to allow qemu and cloud-hypervisor to work with `ctr`.
+
+```
+[plugins.cri.containerd.runtimes.kata-qemu]
+  runtime_type = "io.containerd.kata-qemu.v2"
+  privileged_without_host_devices = true
+  pod_annotations = ["io.katacontainers.*"]
+  [plugins.cri.containerd.runtimes.kata-qemu.options]
+    ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-qemu.toml"
+[plugins.cri.containerd.runtimes.kata-clh]
+  runtime_type = "io.containerd.kata-clh.v2"
+  privileged_without_host_devices = true
+  pod_annotations = ["io.katacontainers.*"]
+  [plugins.cri.containerd.runtimes.kata-clh.options]
+    ConfigPath = "/opt/kata/share/defaults/kata-containers/configuration-clh.toml"
+```
+
+In addition, containerd expects the binary to be in `/usr/local/bin` so add 
+this small script so that it redirects to be able to use either qemu or
+cloud-hypervisor with Kata.
+
+```bash
+$ echo '#!/bin/bash' | sudo tee /usr/local/bin/containerd-shim-kata-qemu-v2
+$ echo 'KATA_CONF_FILE=/opt/kata/share/defaults/kata-containers/configuration-qemu.toml /opt/kata/bin/containerd-shim-kata-v2 $@' | sudo tee -a /usr/local/bin/containerd-shim-kata-qemu-v2
+$ sudo chmod +x /usr/local/bin/containerd-shim-kata-qemu-v2
+$ echo '#!/bin/bash' | sudo tee /usr/local/bin/containerd-shim-kata-clh-v2
+$ echo 'KATA_CONF_FILE=/opt/kata/share/defaults/kata-containers/configuration-clh.toml /opt/kata/bin/containerd-shim-kata-v2 $@' | sudo tee -a /usr/local/bin/containerd-shim-kata-clh-v2
+$ sudo chmod +x /usr/local/bin/containerd-shim-kata-clh-v2
+```
 
-```sh
-$ sudo docker run -it --runtime=kata-runtime --cap-add=IPC_LOCK --cap-add=SYS_ADMIN --device=/dev/vfio/90 -v /dev:/dev -v ${QAT_CONF_LOCATION}:/etc openssl-qat-engine bash
+After the OpenSSL image is built and imported into containerd, a QAT virtual 
+function exposed in the step above can be added to the `ctr` command. Make 
+sure to change the `/dev/vfio` number to one that actually exists on the host 
+system. When using the `ctr` tool, the`configuration.toml` for Kata needs to 
+point to the custom Kata kernel and rootfs built above and the QAT modules in 
+the Kata rootfs need to load at boot. The following steps assume that 
+`kata-deploy` was used to install Kata and qemu is being tested. If using a 
+different hypervisor, different install method for Kata, or a different 
+Intel® QAT chipset then the command will need to be modified. The following 
+was tested with [containerd v1.3.9](https://github.com/containerd/containerd/releases/tag/v1.3.9).
+
+```bash
+$ config_file="/opt/kata/share/defaults/kata-containers/configuration-qemu.toml"
+$ sudo sed -i "/kernel =/c kernel = "\"${KATA_ROOTFS_LOCATION}/${KATA_KERNEL_NAME}\""" $config_file
+$ sudo sed -i "/image =/c image = "\"${KATA_KERNEL_LOCATION}/kata-containers.img\""" $config_file
+$ sudo sed -i "/kernel_params = "\""\"/c kernel_params = "\"modules-load=usdm_drv,qat_c62xvf\""" $config_file
```

Comment:
A few more thoughts:

- I wonder if these search patterns should be anchored (`^kernel`, *etc*) to make them safer?
- Random thought: rather than `sed` you could use `crudini` as it's clearer what's happening (although you'd need to add a comment to tell users to install it using their package manager:
  ```bash
  $ kernel_path="${KATA_ROOTFS_LOCATION}/${KATA_KERNEL_NAME}"
  $ sudo crudini --set "$config_file" hypervisor.qemu kernel "\"$kernel_path\""
  ```
- The result of that `kernel_params` call could be surprising as you're setting it, not appending (aka users may lose kernel params as a result). I'd make it append as shown in https://github.com/kata-containers/kata-containers/blob/main/docs/Developer-Guide.md#enable-full-debug.